controllers/activemqartemis_controller.go

```diff
@@ -333,12 +333,47 @@ func validateExposeModes(customResource *brokerv1beta1.ActiveMQArtemis) (*metav1
 			return &metav1.Condition{
 				Type:    brokerv1beta1.ValidConditionType,
 				Status:  metav1.ConditionFalse,
-				Reason:  brokerv1beta1.ValidConditionFailedConsoleWithInvalidExposeMode,
+				Reason:  brokerv1beta1.ValidConditionFailedInvalidExposeMode,
 				Message: ".Spec.Console has invalid expose mode route, it is only supported on OpenShift",
 			}, false
 		}
 	}
 
+	for _, acceptor := range customResource.Spec.Acceptors {
+		if acceptor.Expose && (acceptor.ExposeMode != nil && *acceptor.ExposeMode == brokerv1beta1.ExposeModes.Ingress || !isOpenshift) &&
+			customResource.Spec.IngressDomain == "" && acceptor.IngressHost == "" {
+			return &metav1.Condition{
+				Type:    brokerv1beta1.ValidConditionType,
+				Status:  metav1.ConditionFalse,
+				Reason:  brokerv1beta1.ValidConditionFailedInvalidIngressSettings,
+				Message: fmt.Sprintf(".Spec.Acceptors %q has invalid ingress settings, IngressHost unspecified and no Spec.IngressDomain default domain provided", acceptor.Name),
+			}, false
+		}
+	}
+
+	for _, connector := range customResource.Spec.Connectors {
+		if connector.Expose && (connector.ExposeMode != nil && *connector.ExposeMode == brokerv1beta1.ExposeModes.Ingress || !isOpenshift) &&
+			customResource.Spec.IngressDomain == "" && connector.IngressHost == "" {
+			return &metav1.Condition{
+				Type:    brokerv1beta1.ValidConditionType,
+				Status:  metav1.ConditionFalse,
+				Reason:  brokerv1beta1.ValidConditionFailedInvalidIngressSettings,
+				Message: fmt.Sprintf(".Spec.Connectors %q has invalid ingress settings, IngressHost unspecified and no Spec.IngressDomain default domain provided", connector.Name),
+			}, false
+		}
+	}
+
+	console := customResource.Spec.Console
+	if console.Expose && (console.ExposeMode != nil && *console.ExposeMode == brokerv1beta1.ExposeModes.Ingress || !isOpenshift) &&
+		customResource.Spec.IngressDomain == "" && console.IngressHost == "" {
+		return &metav1.Condition{
+			Type:    brokerv1beta1.ValidConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  brokerv1beta1.ValidConditionFailedInvalidIngressSettings,
+			Message: ".Spec.Console has invalid ingress settings, IngressHost unspecified and no Spec.IngressDomain default domain provided",
+		}, false
+	}
+
 	return nil, false
```

Comment:
What if console/acceptor/connector.Expose = false and user specified the **ExposeMode** ? should be valid such conditions?

Reply:
I'll add a note in the field documentation to pointed that out.

Reply:
ok.